File: src/RuleService/src/main/java/com/orchestrator/rules/model/Wallet.java
package com.orchestrator.rules.model;

import java.math.BigDecimal;

public class Wallet {
    private BigDecimal balance;
    private String currency; // "CNY", "USD"
    private String status;   // "Active", "Locked"

    public BigDecimal getBalance() { return balance; }
    public void setBalance(BigDecimal balance) { this.balance = balance; }

    public String getCurrency() { return currency; }
    public void setCurrency(String currency) { this.currency = currency; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
}

File: src/WalletService/Models/Wallet.cs
using Orchestrator.Shared.Models;

namespace WalletService.Models;

public class Wallet
{
    public int Id { get; set; }
    public string MembershipId { get; set; } = string.Empty;
    public decimal Balance { get; set; } = 0;
    public Currency Currency { get; set; } = Currency.CNY;
    public WalletStatus Status { get; set; } = WalletStatus.Unlocked;
}
